### PR TITLE
chore: App Router - remove references to `/pages` for remaining pages in `/auth`, `/insights`, `/d`, `signup` and add missing default `/future` page

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4243, CAL-4320, CAL-4321, CAL-4322, CAL-4323, CAL-4324, CAL-4337, CAL-4240, CAL-4241
- Removes references to /pages for remaining pages in /insights, /d, /signup, /auth, /reschedule
- Add the missing default `/future` page
- Fix routing forms pages in app router

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Navigate to:

1. `future/auth/sso/[provider]`
2. `future/auth/sso/direct`
3. `future/auth/signin`
4. `future/auth/setup`
5. `future/signup`
6. `future/d/[link]/[slug]`
7. `future/insights`
8. `future/`
9. `future/routing-forms`
10. `future/routing-forms/[...pages]`
11. `/auth/sso/[provider]`
12. `/auth/sso/direct`
13. `/auth/signin`
14. `/auth/setup`
15. `/signup`
16. `/d/[link]/[slug]`
17. `/insights`
18. `/routing-forms`
19. `/routing-forms/[...pages]`

### CODE REVIEW

File: apps/web/app/future/routing-forms/page.tsx [hunk 1]
```diff
@@ -1,30 +1,7 @@
...
-  return redirect(`/apps/routing-forms/${pages.length ? pages.join("/") : ""}`);
-};
-
-const Page = async ({ params, searchParams }: PageProps) => {
-  const legacyCtx = buildLegacyCtx(headers(), cookies(), params, searchParams);
-  await getPageProps(legacyCtx);
-
-  return null;
+const Page = () => {
+  redirect("/apps/routing-forms/forms");
```
Comment (joeauyeung) [inline apps/web/app/future/routing-forms/page.tsx:4]: Heads up redirecting to this future route throws a client side error.

**Review by joeauyeung — DISMISSED** [review]
Going to the future route of `/apps/routing-forms/forms` throws a client side error. I'm not sure if you want to fix that here or a follow up PR. I'll still approve anyways to unblock you.

File: apps/web/app/future/auth/signin/page.tsx
[the remark this answers is not in the document]
Comment (hbjORbj) [reply apps/web/app/future/auth/signin/page.tsx:1]: I will create a linear ticket.

[the remark this answers is not in the document]
Comment (hbjORbj) [reply apps/web/app/future/auth/signin/page.tsx:1]: ticket ID: CAL-4336

File: apps/web/app/future/routing-forms/page.tsx (hunk 1 above)
Comment (hbjORbj) [reply apps/web/app/future/routing-forms/page.tsx:4]: Thanks, Joe, I pushed a fix!

**Review by joeauyeung — CHANGES_REQUESTED** [review]
Heads up that in both the pages and app route of `/apps/routing-forms/forms` I'm getting the following error. It looks like it's coming from `apps/web/lib/apps/[slug]/[...pages]/getServerSideProps.ts`

CleanShot 2024-09-16 at 11 24 25@2x

**Review by joeauyeung — APPROVED** [review]
Talked with @hbjORbj and we're going to tackle the routing forms pages in another PR.

**Pull request merged**